hazelcast/src/test/java/com/hazelcast/internal/metrics/impl/DynamicMetricsCollectionTest.java

```diff
@@ -180,8 +180,13 @@ public void testDynamicProviderExceptionsAreNotPropagated() {
 
         });
 
-        metricsRegistry.collect(collectorMock);
-        // we just expect there is no exception
+        // we just expect there is no exception apart from AssertionError,
+        // which is for testing only
+        try {
+            metricsRegistry.collect(collectorMock);
```

Comment:
Needs a `fail()` after this statement, otherwise test will pass even when `metricsRegistry.collect` does not throw `AssertionError`. Also, annotate the test with `@RequireAssertEnabled` to ensure assertions are enabled.

Reply:
Added, thanks :+1: